packages/api/app/Controllers/Http/TechnologyController.js

```diff
@@ -290,23 +291,11 @@ class TechnologyController {
 			await technology.locations().detach(null, null, trx);
 		}
 
-		const locationsMap = locations.reduce(
-			(obj, currentLocation) => {
-				const { location_id, location_type } = currentLocation;
-				obj.ids.push(location_id);
-				obj.locations[location_id] = location_type;
-				return obj;
-			},
-			{ ids: [], locations: {} },
-		);
-
-		await technology.locations().attach(
-			locationsMap.ids,
-			(row) => {
-				// eslint-disable-next-line no-param-reassign
-				row.location_type = locationsMap.locations[row.location_id];
-			},
-			trx,
+		await TechnologyLocation.createMany(
+			locations.map((location) => ({
+				technology_id: technology.id,
+				...location,
```

Comment:
Assim sempre sobrescreveremos o `technology_id`, mesmo que já exista
```suggestion
				...location,
				technology_id: technology.id,
```